test/e2e/setup-kubeflex.sh

```diff
@@ -24,6 +24,29 @@ SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)
 :
 
 kind create cluster --name kubeflex --config ${SCRIPT_DIR}/kind-config.yaml
+
+# Install Helm if not available
+if ! command -v helm &> /dev/null; then
+    curl -fsSL -o /tmp/get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3
+    chmod 700 /tmp/get_helm.sh
+    /tmp/get_helm.sh
+fi
+
+# Install ingress-nginx using Helm
+helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+helm install ingress-nginx ingress-nginx/ingress-nginx \
+  --namespace ingress-nginx \
+  --create-namespace \
+  --version 4.10.0 \
+  --set controller.service.type=NodePort \
+  --set controller.hostPort.enabled=true
```

Comment:
the script installs ingress-nginx twice: once via Helm and once via kubectl apply. This may lead to race conditions or resource conflicts. Use only one method to install ingress-nginx.